pkg/azuredisk/azuredisk_v2.go

```diff
@@ -114,9 +141,27 @@ func (d *DriverV2) Run(endpoint, kubeconfig string, testBool bool) {
 		csi.NodeServiceCapability_RPC_GET_VOLUME_STATS,
 	})
 
+	// Start the controllers if this is a controller plug-in
+	if *isControllerPlugin {
+		go controller.StartControllersAndWaitForExit(d.kubeConfig, &d.azVolumeClient, "default", d.objectNamespace)
+	}
+
+	// Register the AzDriverNode
+	if *isNodePlugin {
+		RegisterAzDriverNodeOrDie(d.NodeID, d.kubeClient, d.azVolumeClient, "default", d.objectNamespace)
+	}
+
+	// Start the CSI endpoint/server
 	s := csicommon.NewNonBlockingGRPCServer()
-	// Driver d act as IdentityServer, ControllerServer and NodeServer
+	// Driver d acts as IdentityServer, ControllerServer and NodeServer
 	s.Start(endpoint, d, d, d, testBool)
```

Comment:
Consider adding a `*sync.WaitGroup` parameter to the controller and gRPC server start methods to coordinate the clean shutdown of all of them together.